viewing_party/party.py

```diff
@@ -1,23 +1,213 @@
+    return user_watched_dict
+
+def add_to_watchlist(user_watchlist_dict, movie_info):
+    user_watchlist_dict["watchlist"] = [movie_info]
+    return user_watchlist_dict
+
+def watch_movie(single_user_data,movie_title):    
+    list_of_watchlist_movie_info=single_user_data["watchlist"]
+    list_of_watched_movie_info=single_user_data["watched"]
+
+    #Adds movie info to list of watched movies if it is in the watchlist and then removes the movie info from the watchlist
+    for index,movie_info in enumerate(list_of_watchlist_movie_info):
+        if movie_title == movie_info["title"]:
+            list_of_watched_movie_info.append(movie_info)
+            list_of_watchlist_movie_info.pop(index)
+    return single_user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(single_user_data):
+    list_of_watched_movie_info=single_user_data["watched"]
+    
+    #Calculates the number of watched movies
+    number_of_watched_movies=len(list_of_watched_movie_info)
+
+    list_of_movie_ratings=[]
+    sum_of_all_ratings=0
+
+    if len(list_of_watched_movie_info) == 0:
+        return 0.0
+
+    #Adds rating of each watched movie to a list
+    for movie_info in list_of_watched_movie_info:
+        list_of_movie_ratings.append(movie_info["rating"])
+
+    #Determines the sum of all ratings
+    for rating in list_of_movie_ratings:
+        sum_of_all_ratings+=rating
+
+    #Calculates the average rating using the sum of all ratings and the number of movies
+    average_rating=sum_of_all_ratings/number_of_watched_movies
+    return average_rating
+
+def get_most_watched_genre(single_user_data):
+    list_of_watched_movie_info=single_user_data["watched"]
+    list_of_genres=[]
+    genre_counter_dict={}
+    list_of_watched_genre_counts=[]
+
+    if len(list_of_watched_movie_info) == 0:
+        return None
+
+    #Adds genre of each watched movie to a list
+    for movie_info in list_of_watched_movie_info:
+        list_of_genres.append(movie_info['genre'])
+
+    #Adds genre as the key to a dictionary with a value that counts the number of times it is in the genre list
+    for genre in list_of_genres:
+        if not genre in genre_counter_dict:
+            genre_counter_dict[genre] = 1
+        else:
+            genre_counter_dict[genre] += 1
+
+    #Adds genre count values to a list
+    for genre_count_value in genre_counter_dict.values():
+        list_of_watched_genre_counts.append(genre_count_value)
+
+    #Determines the highest genre count value
+    max_watched_genre_count=max(list_of_watched_genre_counts)
+
+    #Returns a genre from the genre:counter dictionary with a counter value that is the same as the highest genre count value
+    for genre,genre_count_value in genre_counter_dict.items():
+        if genre_count_value == max_watched_genre_count:
+            return genre

```

Comment:
ohhh ok interesting approach! gotta ask ;) food for thought: how could we combine these two for loops to together and find the max?

There is a for loop of some kind being run under the hood of the `max` function, so it's not necessarily saving us time or a for loop on its own